### ISSUE TITLE
[FEATURE REQUEST] File chooser mode

### ISSUE THREAD

**Issue by Rolv-Apneseth** [kyoheiu/felix#261]
Just a possibility, but something like [yazi](https://github.com/sxyazi/yazi)'s `--chooser-file` (or `ranger`'s `--choosefiles`) flag would be very useful. This just writes the selected file(s) to a given file path instead of opening them directly. My use case is for using it in a Neovim plugin, such as others I have in my plugin [tfm.nvim](https://github.com/Rolv-Apneseth/tfm.nvim).

**Comment by kyoheiu** [comment]
Hi,
could you let me understand this a little bit more?

I think this is similar to this issue of joshuto https://github.com/kamiyaa/joshuto/issues/96 .
This feature would enable `fx --chosefiles=/file/path` command, and after the launch, the path of a file executed in felix would be added to `/file/path` text file with `\n`, right?
For example, if I open `~/.config/felix/config.yaml` with `--choosefiles=~/sample.txt` option, the full path of `config.yaml` would be written to `~/sample.txt`.

I'd be appreciate if you teach me how this would work with the vim plugin.

**Comment by Rolv-Apneseth** [comment]
Sure yeah.

So for example when run with `felix --choosefiles=~/sample.txt`, I would want it to open felix normally, but when selecting a file, instead of opening it, just write it's path to `~/sample.txt`, so that the selected file can be opened by something else. Hence it's just being used in a kind of file picker/chooser mode.

As a side note, is there currently a way of passing just a file path and having that file path focused when opening `felix`? Either it's own flag  or just accepting any path instead of just directories. `yazi` just accepts the path to the file, and `ranger` has a `--selectfile` flag.

**Comment by Rolv-Apneseth** [comment]
By the way - very cool project, thanks for making it

**Comment by kyoheiu** [comment]
Sorry for the late reaction.
#262 implements `--choosefiles=/path/to/textfile` option.
It'be great if you check the behavior.

**Comment by Rolv-Apneseth** [comment]
No problem, thanks for implementing this!

Checked it out and just 2 things (my opinion):

1. If the file doesn't exist, it should just create it
2. I believe it would be better to close the program once a file is selected. That's how other terminal file managers tend to do it

The use case for me is:
- Open `felix` inside `neovim`, and have it focus the file that was being edited (this is where that other feature would be useful)
- Select a different file from within `felix`, which writes this path to a file and closes, bringing you back to `neovim`
-  A plugin can then read that path and open the file in `neovim`

**Comment by RioNight** [comment]
I think it be more ergonomic to first select a bunch of files in fx and enter an `fx` command to write selected files to a user-provided path. That way you don't have to manually type `fx --choosefiles="/some/path"`.
Then you can ask the user to quit after writing to the path (or have an option in a config called `exit-after-choosefiles`.

**Comment by Rolv-Apneseth** [comment]
Could be an option I guess, though hopefully `--choosefiles` would still be available

**Comment by kyoheiu** [comment]
> * If the file doesn't exist, it should just create it
> * I believe it would be better to close the program once a file is selected. That's how other terminal file managers tend to do it

@Rolv-Apneseth 
Fixed in this way.

**Comment by Rolv-Apneseth** [comment]
Awesome, thank you.

Would it make more sense to override the file with the new path rather than just append the path to the existing file? It wouldn't make a difference for my use case though so for me it's fine as it is.

One more potential feature for this, how about being able to output multiple files with linewise visual mode? Absolutely not necessary but it would be nice. This could potentially be a separate feature, as, correct me if I'm wrong, but I don't think there's a way to open multiple files to edit at the same time right now?

**Comment by kyoheiu** [comment]
OK, I'll fix it to open in non-append mode.
And about the linewise visual mode, it could be easy to add the feature to "open" multiple files with choosefiles option.
In other options e.g. just `fx`, though, I might disable that feature because felix is just a tui file manager, and while it's completely normal to, e.g., open multiple images with macOS' preview, terminals are not created for that use case IMO.

**Comment by kyoheiu** [comment]
@Rolv-Apneseth 
Ok, fixed. Now
- `--choosefiles` option always create or truncate new text file.
- In linewise visual mode, if `--choosefiles` option is on, you can output multiple file paths by `Enter`, `l` or right arrow key.

**Comment by Rolv-Apneseth** [comment]
That's awesome, works great!

**Comment by kyoheiu** [comment]
Merged to `develop`. It'll be published to crate.io and other package repositories after adding some other features.

**Comment by Rolv-Apneseth** [comment]
Nice, thanks for implementing this. I'll also subscribe to #263 and once that's done I'll be sure to include `felix` in [tfm.nvim](https://github.com/Rolv-Apneseth/tfm.nvim.git)

**Comment by kyoheiu** [comment]
#263 implemented in #272.

**Comment by Rolv-Apneseth** [comment]
Yep fair enough, no need to do anything else if you don't want. I completely understand.

As for the plugin - no idea how many use it. Probably not all that many.

I will say though, these features are nice to have for being able to easily integrate this file manager into other programs, and the other major terminal file managers have these options. Realistically, if you want to leave it here, it is good enough as it is since `choosefiles` works and selecting the file is just a nice-to-have.

**Comment by lukeflo** [comment]
Will the `--choosefiles` option be merged into `main`?  BTW interesting project!

**Comment by kyoheiu** [comment]
At least until I understand what is the minimum to meet the demand, no, it won't.
But I'll keep this issue open.
